identity/app/controllers/SaveContentController.scala

```diff
@@ -79,32 +75,31 @@ class SaveContentController @Inject() ( api: IdApiClient,
     }
   }
 
-
-  def listSavedContent = authenticatedActions.authAction.async { implicit request  =>
+  def listSavedContentPage() = authenticatedActions.authAction.async { implicit request =>
 
     val idRequest = identityRequestParser(request)
+    val pageNum = idRequest.page.getOrElse(1)
 
     savedArticleService.getOrCreateArticlesList(request.user.auth).flatMap {
       case Right(savedArticles) =>
-        val form = savedArticlesForm.fill(SavedArticleData(savedArticles.newestFirst.map(_.shortUrl)))
-        fillFormWithApiDataAndGetResult(idRequest, form, savedArticles)
-       case Left(errors) =>
+        fillFormWithApiDataForPageAndGetResult(idRequest, savedArticles, pageNum)
+      case Left(errors) =>
         val formWithErrors = errors.foldLeft(savedArticlesForm) {
           case (formWithErrors, Error(message, decription, _, context)) =>
             formWithErrors.withError(context.getOrElse(""), message)
         }
-        Future.successful(NoCache(Ok(views.html.profile.savedForLater(page, formWithErrors, List.empty, formActionUrl(idUrlBuilder, idRequest, "/saved-for-later")))))
+        val pageData = pageDataBuilder.apply(List.empty, emptyArticles, idRequest, pageNum)
```

Comment:
Why are you calling `apply` here? Can you not just call `pageDataBuilder` directly, like a function?